### ISSUE TITLE
Alternates API

### ISSUE THREAD

**Issue by carlosmn** [libgit2/libgit2#2621]
If a repository/odb has an alternates file, we follow it to grab objects from other repositories. But we don't have any way of interacting with it from a program.

It is "just" a newline-delimited list of paths in a file so we might not necessarily want to do too much in libgit2, but maybe it can be abstracted just enough so non-default backends can also make use of it.

So I was thinking of maybe having

``` C
git_odb_alternates_read(git_strarray*, git_odb *);
git_odb_alternates_write(git_odb *, git_strarray *);
git_odb_alternates_append(git_odb *, const char *);
```

with the appropriate additions to the odb backend interface, which should make implementing things like `git clone --reference` or `--shared` that much easier.

**Comment by ethomson** [comment]
Yeah, I think I agree.  I don't think that it makes much sense to have this on the ODB layer (at least to match git semantics).  Alternates are queried before looking at _any_ of our ODBs.  This makes it seem like it's an alternate to _an_ ODB, when alternates are really an alternative to _all our_ ODBs.
